### PR TITLE
Implemented Button from scss

### LINKED ISSUE #2560: (SP: 5) Create AppButton component aligned with design system
#### Description:
Implement a reusable `AppButton` component that adheres to the design system standards. This component will be used throughout the application to maintain consistency in button styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=19219-83444&t=Bm41HUUhKFERVBCG-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  - Support different button variants (e.g., primary, tonal text).
  - Ensure hover, focus, active, and disabled states are defined according to the design system.
  
- **Props:**
  - `variant`: Defines button type (primary, tonal, text).
  - `size`: Supports different sizes (xs, sm, md, lg).
  - `color`: Supports different colors (primary, secondary, error, success)
  - `loading`: Disables the button when true. When loading, the button shouldn't shrink or change its size compared to its default state
  - `disabled`: Disables the button when true.

### PR DESCRIPTION
**Pull request by Mav-Ivan**
- Implemented Button 
- Restructured for the design system

### CODE REVIEW

File: src/design-system/components/button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { forwardRef, ReactNode } from 'react'
+import {
+  Button as MuiButton,
+  CircularProgress,
+  ButtonProps as MuiButtonProps
+} from '@mui/material'
+
+import { cn } from '~/utils/cn'
+
+import './Button.scss'
```
Comment (dudchakk) [inline src/design-system/components/button/Button.tsx]: add alias for the import

File: src/design-system/scss/_variables.scss [hunk 2]
```diff
@@ -461,10 +461,10 @@ $box-shadow-md:
   0px 8px 16px 0px rgba(0, 0, 0, 0.1) !default;
 $box-shadow-lg:
   0px 0px 8px 0px rgba(0, 0, 0, 0.08),
-  0px 10px 20px 0px rgba(0, 0, 0, 0.1) !default;
+  0px 10px 12px 0px rgba(0, 0, 0, 0.1) !default;
 $box-shadow-xl:
-  0px 0px 10px 0px rgba(0, 0, 0, 0.08),
-  0px 12px 24px 0px rgba(0, 0, 0, 0.1) !default;
+  0px 0px 14px 0px rgba(0, 0, 0, 0.08),
+  0px 11px 18px 0px rgba(0, 0, 0, 0.1) !default;
```
Comment (dudchakk) [inline src/design-system/scss/_variables.scss:467]: why did you change the shadows here?

File: src/design-system/components/button/Button.tsx (hunk 1 above)
Comment (Mav-Ivan) [reply src/design-system/components/button/Button.tsx]: fixed

File: src/design-system/scss/_variables.scss (hunk 2 above)
Comment (Mav-Ivan) [reply src/design-system/scss/_variables.scss:467]: thats more corresponding to the mockup

**Review by dudchakk — APPROVED** [review]

**Review by ipasic-softserve — APPROVED** [review]

**Review by Mike-Popovych — APPROVED** [review]
